Don't set newer tile timestamp when invalidating

Helium314 opened the pull request:
fixes #5068

Tested by logging tile timestamps before and after invalidating: works as intended.

westnordost commented on app/src/androidTest/java/de/westnordost/streetcomplete/data/download/tiles/DownloadedTilesDaoTest.kt:
```diff
@@ -67,7 +67,7 @@ class DownloadedTilesDaoTest : ApplicationDbTestCase() {
 
     @Test fun updateTime() {
```
Should update the names of the test functions too.

westnordost commented on app/src/main/java/de/westnordost/streetcomplete/data/download/tiles/DownloadedTilesDao.kt:
```diff
@@ -31,16 +31,20 @@ class DownloadedTilesDao(private val db: Database) {
 
-    fun updateAllTimes(time: Long) {
-        db.update(NAME, values = listOf(DATE to time))
+    fun updateAllTimesIfNewer(time: Long) {
```
Sounds odd. Wouldn't `updateAllTimesNewerThan` be more descriptive? (Same for the other function, `updateTimeIfNewerThan` or something.)

westnordost reviewed (changes requested):
Since the function does something different now, it would make sense to add two tests that test the new behavior: I.e. that only the times are updated that are older than X.

Helium314 commented:
I renamed the functions and added a test.
But for whatever reason, I can't run the android tests on my device, so I can't actually check whether the test succeeds...
